crates/store/re_sorbet/src/column_descriptor_ref.rs

```diff
@@ -0,0 +1,94 @@
+use arrow::datatypes::{DataType as ArrowDatatype, Field as ArrowField, Fields as ArrowFields};
+
+use re_log_types::EntityPath;
+
+use crate::{
+    ColumnDescriptor, ComponentColumnDescriptor, IndexColumnDescriptor, RowIdColumnDescriptor,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum ColumnDescriptorRef<'a> {
+    RowId(&'a RowIdColumnDescriptor),
+    Time(&'a IndexColumnDescriptor),
+    Component(&'a ComponentColumnDescriptor),
+}
+
+impl ColumnDescriptorRef<'_> {
+    #[inline]
+    pub fn entity_path(&self) -> Option<&EntityPath> {
+        match self {
+            Self::RowId(_) | Self::Time(_) => None,
+            Self::Component(descr) => Some(&descr.entity_path),
+        }
+    }
+
+    #[inline]
+    pub fn short_name(&self) -> String {
+        match self {
+            Self::RowId(descr) => descr.name().to_owned(),
+            Self::Time(descr) => descr.timeline.name().to_string(),
+            Self::Component(descr) => descr.component_name.short_name().to_owned(),
+        }
+    }
+
+    #[inline]
+    pub fn is_static(&self) -> bool {
```

Comment:
I know the existing `ColumnDescriptor` already had these functions, and that they are undocumented. But do we really need them? I prefer to remove them, as I find it hard to explain why the row-id column is not static, for instance.

If we do want them, then let us add docstrings for why their indended use is

Reply:
I'll remove all the ones I dont need (which is all but `short_name`)